scene/2d/animated_sprite.cpp

```diff
@@ -453,6 +453,8 @@ void AnimatedSprite::_notification(int p_what) {
 				ofs -= s / 2;
 
 			if (Engine::get_singleton()->get_snap_2d_transforms()) {
+				ofs = ofs.round();
+			} else if (Engine::get_singleton()->get_use_pixel_snap()) {
 				ofs = ofs.floor();
```

Comment:
Is the current documentation clear enough about how 2d transform and pixel snapping will behave differently, including this new difference? (And the fact that 2d transform has precedence.)

Reply:
To be honest I'm not convinced that we even need the `floor` with the pixel snap (which is actually done in the vertex shader). But I didn't want to remove it in this PR, just in case there is an actual need for it, we can evaluate removing it in a separate PR.

Perhaps I should take the opportunity to rename pixel_snap, at least internally because it seems to be causing confusion. The name is historical, but really `vertex_shader_snap` or something might be better. Or the name reduz used in his PR I will check that.

In either case I don't think mentioning it in the docs is wise at this stage until it is finalized. `snap_2d_transforms` does not totally take precedence, only here on CPU. On the GPU, `pixel_snap` still acts alongside the CPU snapping.